pkg/gateway/db/db.go

```diff
@@ -0,0 +1,22 @@
+package db
+
+import (
+	"context"
+)
+
+var driver Driver
+
+// Register makes an database driver available
+func Register(d Driver) {
+	if d == nil {
+		panic("gateway/db: Register driver is nil")
+	}
+	driver = d
+}
```

Comment:
I think we can assume gateway only program towards postgres and remove this layer.